### DISCUSSION TITLE
Setting --encrypt as a default option for add

### DISCUSSION THREAD

**Discussion by breisfeld** [twpayne/chezmoi#3469] [Q&A] [answered]
Hi, Is there a way to set the default ```add``` to encrypt such that the ```--encrypt``` option is not needed?
I didn't see a way to set this in the config file, but I may have missed it.

Thanks!

**Comment by twpayne** [comment, marked as answer]
There's currently no way to do this. Do you _really_ want to encrypt every file managed by chezmoi?

As a work-around, you can create a shell alias, something like:

```shell
alias chezmoi-add="chezmoi add --encrypt"
```

and then use `chezmoi-add` instead of `chezmoi add`.

Alternatively, you could define a plugin:

```shell
#!/bin/bash
# put this in a file called chezmoi-adde somewhere in your path
exec chezmoi add --encrypt "$*"
```

and then use `chezmoi adde` instead of `chezmoi add`.

**Reply by breisfeld** [reply]
Thank you for the great advice! 

I wanted encryption as the default, since I anticipate forgetting that I have sensitive information included in a file I wish to ```add```.

**Reply by twpayne** [reply]
OK. #3470 adds a configuration variable so you can set `add.encrypt` in your configuration file instead of specifying `--encrypt`.

**Reply by breisfeld** [reply]
Wow! You  are awesome!

**Reply by twpayne** [reply]
https://github.com/twpayne/chezmoi/pull/3472 adds support for scanning any files you add for secrets.
